### PR TITLE
Adds redis-cli preferences

### PR DESCRIPTION
**Pull request by itamarhaber**

### CODE REVIEW

**Comment by laixintao** [comment]
Is there an exmaple file like http://download.redis.io/redis-stable/redis.conf for `.redisclirc` :) ?

**Comment by itamarhaber** [comment]
Good one - will mark that as a TODO due to the complexity involved in covering all edge cases :P

**Comment by itamarhaber** [comment]
@simonprickett please review this

File: topics/rediscli.md [hunk 1]
```diff
@@ -345,6 +345,22 @@ key, like in the following example:
     127.0.0.1:6379> ZADD<TAB>
     127.0.0.1:6379> ZCARD<TAB>
 
+Once you've entered a valid command name at the prompt, its syntax will be
```
Comment (simonprickett) [inline topics/rediscli.md:348]: Maybe:

"Once you've typed a Redis command name at the prompt, the CLI will display syntax hints.  This behavior can be turned on and off via the CLI preferences."

File: topics/rediscli.md [hunk 2]
```diff
@@ -345,6 +345,22 @@ key, like in the following example:
...
+displayed by the CLI by default. The default syntax hints display can be
+controlled via the CLI's preferences.
+
+## Preferences
+
+The CLI's behavior can be customized to your preferences. Once started, the CLI
```
Comment (simonprickett) [inline topics/rediscli.md:354]: Perhaps:

"There are two ways to customize the CLI's behavior.  The file `.redisclirc` in your home directory is loaded by the CLI on startup.  Preferences can also be set during a CLI session, in which case they will last only the the duration of the session."

File: topics/rediscli.md [hunk 3]
```diff
@@ -345,6 +345,22 @@ key, like in the following example:
...
+loads the preferences from a file named `.redisclirc` in the home directory of
+the user. Preferences can also be set during runtime.
+
+To set a preference, either via file or in runtime, use the the special `:set`
```
Comment (simonprickett) [inline topics/rediscli.md:358]: Perhaps: "To set preferences, use the special `:set` command.  The following preferences can be set, either by typing the command in the CLI or adding it to the `.redisclirc` file:"

File: topics/rediscli.md (hunk 1 above)
Comment (itamarhaber) [reply topics/rediscli.md]: Accepted

File: topics/rediscli.md (hunk 3 above)
Comment (itamarhaber) [reply topics/rediscli.md]: Accepted

File: topics/rediscli.md (hunk 2 above)
Comment (itamarhaber) [reply topics/rediscli.md]: Accepted

**Pull request merged**